internal/extsvc/gitlab/merge_requests.go

```diff
@@ -52,18 +52,21 @@ type MergeRequest struct {
 	Pipelines []*Pipeline
 }
 
+// SetWIP ensures a "WIP:" prefix on the given title. If a "Draft:" prefix is found, that one is retained instead.
 func SetWIP(title string) string {
+	if strings.HasPrefix(title, "Draft:") {
+		return title
+	}
 	if !strings.HasPrefix(title, "WIP:") {
 		return "WIP: " + title
 	}
 	return title
 }
```

Comment:
Tiniest nitpick of all time: I'd make the checks the same.

```suggestion
	if strings.HasPrefix(title, "Draft:") {
		return title
	}
	if strings.HasPrefix(title, "WIP:") {
		return title
	}
	return "WIP: " + title
}
```